### PR TITLE
Adding more libs to openshift and adding default yaml's

### PR DESCRIPTION
**Pull request by vavuthu**

### CODE REVIEW

File: oc/openshift_ops.py [hunk 1]
```diff
@@ -184,3 +185,165 @@ def create_project(self, project):
...
+                cephblockpool_name_prefix="autotests-blockpool",
+                failureDomain="host",
+                size=3)
+
+        """
+        service_cbp = self.dyn_client.resources.get(
+            api_version='v1',
+            kind='CephBlockPool'
+        )
+        cephblockpool_name = "%s-%s" % (
```
Comment (petr-balogh) [inline oc/openshift_ops.py]: Please use new formatting f"{cephblockpool_name_prefix}-{get_random_str()}"

File: oc/openshift_ops.py [hunk 2]
```diff
@@ -184,3 +185,165 @@ def create_project(self, project):
...
+        Raises:
+            KeyError when error occured
+
+        Examples:
+            create_cephblockpool(
+                namespace,
+                cephblockpool_name_prefix="autotests-blockpool",
+                failureDomain="host",
+                size=3)
+
```
Comment (petr-balogh) [inline oc/openshift_ops.py]: You can delete empty line here

File: oc/openshift_ops.py [hunk 3]
```diff
@@ -184,3 +185,165 @@ def create_project(self, project):
...
+        **kwargs
+    ):
+        """
+        Creates cephblock pool
+
+        Args:
+            namespace (str): namespace to create cephblockpool
+            cephblockpool_name_prefix (str): prefix given to cephblockpool
+
+        kwargs:
```
Comment (petr-balogh) [inline oc/openshift_ops.py]: I would capitalize it as there is all capitalized, and if we will specify Kwargs in docs I would follow the same syntax as for Args, WDYT? If we decide to do it like this we can send another PR to have it in documentation.
Like:
```python
Kwargs:
    failureDomain (str): description
    size (int): description
```

File: oc/openshift_ops.py [hunk 4]
```diff
@@ -184,3 +185,165 @@ def create_project(self, project):
...
+            api_version='v1',
+            kind='CephBlockPool'
+        )
+        cephblockpool_name = "%s-%s" % (
+            cephblockpool_name_prefix,
+            get_random_str()
+        )
+        # overwrite the namespace with openshift-storage, since cephblockpool
+        # is tied-up with openshift-storage
+        namespace = "openshift-storage"
```
Comment (petr-balogh) [inline oc/openshift_ops.py]: "openshift-storage" already defined in [defaults]( https://github.com/red-hat-storage/ocs-ci/blob/master/ocs/defaults.py), please use from there.

File: oc/openshift_ops.py [hunk 5]
```diff
@@ -184,3 +185,165 @@ def create_project(self, project):
...
+                clusternamespace="openshift-storage",
+                fstype="xfs"
+            )
+
+        """
+        service_sc = self.dyn_client.resources.get(
+            api_version='v1',
+            kind='StorageClass'
+        )
+        storageclass_name = "%s-%s" % (sc_name_prefix, get_random_str())
```
Comment (petr-balogh) [inline oc/openshift_ops.py]: same

File: oc/openshift_ops.py [hunk 6]
```diff
@@ -184,3 +185,165 @@ def create_project(self, project):
...
+            create_pvc(
+                namespace,
+                storageclass,
+                accessmode="ReadWriteOnce ReadOnlyMany",
+                pvc_size=5
+            )
+
+        """
+        service_pvc = self.dyn_client.resources.get(
+            api_version='v1',
```
Comment (petr-balogh) [inline oc/openshift_ops.py]: define  and take from defaults

File: oc/openshift_ops.py [hunk 7]
```diff
@@ -184,3 +185,165 @@ def create_project(self, project):
...
+            kind='PersistentVolumeClaim'
+        )
+        pvc_name = "%s-%s" % (pvc_name_prefix, get_random_str())
```
Comment (petr-balogh) [inline oc/openshift_ops.py]: New formatting, same bellow

File: templates/ocs-deployment/cephblockpool.yaml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+---
+apiVersion: {{ rook_api_version }}
+kind: CephBlockPool
+metadata:
+  name: {{ cephblockpool_name | default('ocsci-cephblockpool') }}
+  namespace: {{ cluster_namespace | default('openshift-storage') }}
+spec:
+  failureDomain: {{ failureDomain | default('host') }}
+  replicated:
+    size: {{ size | default('2') }}
```
Comment (petr-balogh) [inline templates/ocs-deployment/cephblockpool.yaml]: missing empty line at the end of line

File: templates/ocs-deployment/pvc.yaml [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ pvc_name | default('ocsci-pvc') }}
+  namespace: {{ cluster_namespace | default('openshift-storage') }}
+spec:
+  accessModes: {{ access_mode }}
+  resources:
+    requests:
+      storage: {{ storage | default('3Gi') }}
+  storageClassName: {{ storageclass_namespace }}
```
Comment (petr-balogh) [inline templates/ocs-deployment/pvc.yaml]: same

File: templates/ocs-deployment/storageclass.yaml [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+kind: StorageClass
+metadata:
+   name: {{ storageclass_name | default('ocsci-storageclass') }}
+provisioner: ceph.rook.io/block
+allowVolumeExpansion: {{ volume_expansion | default('True') }}
+reclaimPolicy: {{ reclaimPolicy | default('Delete') }}
+parameters:
+  blockPool: {{ blockPool }}
+  clusterNamespace: {{ clusterNamespace | default('openshift-storage') }}
+  fstype: {{ fstype | default('xfs') }}
```
Comment (petr-balogh) [inline templates/ocs-deployment/storageclass.yaml]: same

File: oc/openshift_ops.py [hunk 11]
```diff
@@ -184,3 +185,165 @@ def create_project(self, project):
...
+            create_cephblockpool(
+                namespace,
+                cephblockpool_name_prefix="autotests-blockpool",
+                failureDomain="host",
+                size=3)
+
+        """
+        service_cbp = self.dyn_client.resources.get(
+            api_version='v1',
+            kind='CephBlockPool'
```
Comment (RazTamir) [inline oc/openshift_ops.py]: Please create a file to hold the several kinds as strings

File: oc/openshift_ops.py [hunk 12]
```diff
@@ -184,3 +185,165 @@ def create_project(self, project):
...
+        Examples:
+            create_cephblockpool(
+                namespace,
+                cephblockpool_name_prefix="autotests-blockpool",
+                failureDomain="host",
+                size=3)
+
+        """
+        service_cbp = self.dyn_client.resources.get(
+            api_version='v1',
```
Comment (RazTamir) [inline oc/openshift_ops.py]: Let's maintain the api_version in defaults so we will be able to change a single place in the future

File: utility/utils.py [hunk 13]
```diff
@@ -662,3 +664,38 @@ def download_file(url, filename):
...
+        data (dict) : data to replace in the template
+
+    Returns:
+        dict: dictionary format of data
+
+    Example:
+        generate_data("pvc.yaml", pvc_data)
+    """
+    _templating = templating.Templating()
+    template_path = os.path.join(template_dir, template_name)
```
Comment (RazTamir) [inline utility/utils.py]: This implementation is very fragile.
Why not import the template to the relevant file and use it:
in openshift_ops.py import the yaml file and pass it there.
It will work?

File: oc/openshift_ops.py [hunk 14]
```diff
@@ -184,3 +185,165 @@ def create_project(self, project):
...
+        # overwrite the namespace with openshift-storage, since cephblockpool
+        # is tied-up with openshift-storage
+        namespace = "openshift-storage"
+        cephblockpool_data = {}
+        cephblockpool_data['cephblockpool_name'] = cephblockpool_name
+        cephblockpool_data['rook_api_version'] = defaults.rook_api_version
+        cephblockpool_data['failureDomain'] = kwargs.get('failureDomain')
+        cephblockpool_data['size'] = kwargs.get('size')
+
+        data = generate_data("cephblockpool.yaml", cephblockpool_data)
```
Comment (RazTamir) [inline oc/openshift_ops.py]: See my other comment in utils.py about the yaml file.
I think we should pass here the yaml file without the need to manipulating it in the generate_data function cause in case I'm creating a local yaml in this file, it will not work

File: oc/openshift_ops.py [hunk 15]
```diff
@@ -184,3 +185,165 @@ def create_project(self, project):
...
+        storageclass_name = "%s-%s" % (sc_name_prefix, get_random_str())
+
+        sc_data = {}
+        sc_data['k8s_api_version'] = defaults.k8s_api_version
+        sc_data['storageclass_name'] = storageclass_name
+        sc_data['volume_expansion'] = allow_volume_expansion
+        sc_data['reclaimPolicy'] = reclaim_policy
+        sc_data['blockPool'] = kwargs.get('blockPool')
+        sc_data['clusterNamespace'] = kwargs.get('clusterNamespace')
+        sc_data['fstype'] = kwargs.get('fstype')
```
Comment (RazTamir) [inline oc/openshift_ops.py]: I like this way of processing data - +1

File: oc/openshift_ops.py [hunk 16]
```diff
@@ -184,3 +185,165 @@ def create_project(self, project):
...
+
+        data = generate_data("storageclass.yaml", sc_data)
+        service_sc.create(body=data)
+
+        return storageclass_name
+
+    def create_pvc(
```
Comment (RazTamir) [inline oc/openshift_ops.py]: Let's move this to another class VPC

File: oc/openshift_ops.py [hunk 17]
```diff
@@ -184,3 +185,165 @@ def create_project(self, project):
...
+        cephblockpool_data['rook_api_version'] = defaults.rook_api_version
+        cephblockpool_data['failureDomain'] = kwargs.get('failureDomain')
+        cephblockpool_data['size'] = kwargs.get('size')
+
+        data = generate_data("cephblockpool.yaml", cephblockpool_data)
+        service_cbp.create(body=data, namespace=namespace)
+
+        return cephblockpool_name
+
+    def create_storageclass(
```
Comment (RazTamir) [inline oc/openshift_ops.py]: Let's move this to another class StorageClass

File: ocs/defaults.py [hunk 18]
```diff
@@ -6,3 +6,6 @@
 ROOK_CLUSTER_NAMESPACE = 'openshift-storage'
 KUBECONFIG_LOCATION = 'auth/kubeconfig'  # relative from cluster_dir
 CLUSTER_NAME = 'ocs-ci-cluster'
+k8s_api_version = 'storage.k8s.io/v1'
```
Comment (RazTamir) [inline ocs/defaults.py]: Capital letters - same for all new additions

File: utility/utils.py (hunk 13 above)
Comment (vavuthu) [reply utility/utils.py]: I am not sure w can directly import yaml and use it because we need to replace the variables in yaml file.
Other way is load yaml into dict and update dictionary with variable values which is crude way of doing it.

let me know if anyone has better way of rendering yaml file with data.

File: oc/openshift_ops.py (hunk 3 above)
Comment (vasukulkarni) [reply oc/openshift_ops.py]: I think we should separate functions into OCS and OCP,  this(*create_cephblock*) qualifies as OCS function and as such should be moved there and use the OCS way of doing things.

File: utility/utils.py (hunk 13 above)
Comment (vasukulkarni) [reply utility/utils.py]: @vavuthu Loading yaml into dict and updating dict and converting back to yaml is the clean way of edition/modifying the yaml. It will provide great flexibility inside the function to do things and not expose yaml to users at all for simplification,  users will just call the functions and pass the args that they will have to modify.

File: oc/openshift_ops.py (hunk 16 above)
Comment (shylesh) [reply oc/openshift_ops.py]: +1 here. I see a more natural heirarchy of kubernetes volumes . Class KubeVolume() and child classes could be PVC(KubeVolume), RBD(KubeVolume), CephFS(KubeVolume). I haven't really worked on all of these from kubernetes perspective but by looking at kubernetes docs this is what I felt. May be other can shine some light on this.

File: templates/ocs-deployment/cephblockpool.yaml (hunk 8 above)
Comment (vasukulkarni) [inline templates/ocs-deployment/cephblockpool.yaml]: I am thinking this is OSD replica, If so, the default recommended is '3'

File: utility/utils.py (hunk 13 above)
Comment (petr-balogh) [reply utility/utils.py]: @vasukulkarni @RazTamir I think that @vavuthu is trying to use the jinja2 templates from the reason that it's easy to have the file somewhere stored (template folder) with all variables defined where we can use powerful of jinja2 templates like filters (default and others), conditions, loops etc.  And just passing the data in the template and render it and got the data generated from template. If you will just use the pure yaml, there is no such way to use defaults, loops, conditions and so on. So I think that this is the reason why @vavuthu took this way here as we discussed this on some of our mtg.

File: oc/openshift_ops.py [hunk 19]
```diff
@@ -184,3 +185,165 @@ def create_project(self, project):
...
+        """
+        Creates cephblock pool
+
+        Args:
+            namespace (str): namespace to create cephblockpool
+            cephblockpool_name_prefix (str): prefix given to cephblockpool
+
+        kwargs:
+            The keys, values in kwargs are:
+                - failureDomain : (str)
```
Comment (ebenahar) [inline oc/openshift_ops.py]: Please keep in this format:
```python
failureDomain (str): <DESCRIPTION>
size (int): <DESCRIPTION>
```

File: utility/utils.py (hunk 13 above)
Comment (vasukulkarni) [reply utility/utils.py]: @petr-balogh The inital template thing is fine, but we are saying 2 things here
1) Do this operation not in utils but close to the Class function it belongs too
2) Later edit/modify/apply should be class functions rather than template modifications because it will be a small dict.

File: templates/ocs-deployment/cephblockpool.yaml (hunk 8 above)
Comment (ramkiperiy) [reply templates/ocs-deployment/cephblockpool.yaml]: Please change the param name from "size" to "replicasize"

File: templates/ocs-deployment/storageclass.yaml (hunk 10 above)
Comment (nehaberry) [reply templates/ocs-deployment/storageclass.yaml]: @vavuthu  This storageclass.yaml uses the flex provisioner(ceph.rook.io/block). Shouldn't we be adding the CSI provisioner based  storageclass.yaml files  from the rook repo's (csi/examples/rbd and csi/examples/cephfs) . Please comment. Once #58 is merged, we would not need the "ceph.rook.io/block" provisioner based SC files.

File: oc/openshift_ops.py (hunk 11 above)
Comment (vavuthu) [reply oc/openshift_ops.py]: Done

File: oc/openshift_ops.py (hunk 12 above)
Comment (vavuthu) [reply oc/openshift_ops.py]: Done

File: oc/openshift_ops.py (hunk 1 above)
Comment (vavuthu) [reply oc/openshift_ops.py]: Done

File: templates/ocs-deployment/storageclass.yaml (hunk 10 above)
Comment (vavuthu) [reply templates/ocs-deployment/storageclass.yaml]: I will make chnages to use CSI once #58 is merged

File: templates/ocs-deployment/cephblockpool.yaml (hunk 8 above)
Comment (vavuthu) [reply templates/ocs-deployment/cephblockpool.yaml]: Done

File: templates/ocs-deployment/pvc.yaml (hunk 9 above)
Comment (vavuthu) [reply templates/ocs-deployment/pvc.yaml]: Done

File: templates/ocs-deployment/storageclass.yaml (hunk 10 above)
Comment (vavuthu) [reply templates/ocs-deployment/storageclass.yaml]: > same

Done

File: utility/utils.py (hunk 13 above)
Comment (vavuthu) [reply utility/utils.py]: I removed the this function since we have another function ( generate_yaml_from_jinja2_template_with_data ) which was merged recently which served the same purpose. generate_yaml_from_jinja2_template_with_data

File: oc/openshift_ops.py (hunk 14 above)
Comment (vavuthu) [reply oc/openshift_ops.py]: I removed the this function since we have another function ( generate_yaml_from_jinja2_template_with_data ) which was merged recently which served the same purpose. generate_yaml_from_jinja2_template_with_data

File: oc/openshift_ops.py (hunk 16 above)
Comment (vavuthu) [reply oc/openshift_ops.py]: Implemented the same in volume.py

File: oc/openshift_ops.py (hunk 17 above)
Comment (vavuthu) [reply oc/openshift_ops.py]: Done

File: ocs/defaults.py (hunk 18 above)
Comment (vavuthu) [reply ocs/defaults.py]: DONE

[the remark this answers is not in the document]
Comment (vavuthu) [reply ocs/defaults.py]: DONE

File: oc/openshift_ops.py (hunk 19 above)
Comment (vavuthu) [reply oc/openshift_ops.py]: Done

File: utility/utils.py [hunk 20]
```diff
@@ -870,3 +871,15 @@ def update_dict_recursively(d, u):
         else:
             d = {k: u[k]}
     return d
+
+
+def get_random_str(size=13):
+    """
+    generates the random string of 13 characters
+
```
Comment (petr-balogh) [inline utility/utils.py]: This function doesn't return just random 13 chars but whatever size you define, so please change the docstring accordingly. I won't block this PR and merge it now, but plese fix in some next PR @vavuthu 
```python
args:
    size (int)
```

Comment (vavuthu) [reply utility/utils.py]: Done

**Comment by vasukulkarni** [comment]
@shylesh can you syncup with @vavuthu , Some of the functions I see are overlapping here.

File: ocs/volumes.py [hunk 21]
```diff
@@ -0,0 +1,257 @@
...
+    """
+    Class which contains Ceph RBD related functionality
+
+    Attributes:
+        name (str): name of the RBD volume
+        namepsace (str): namespace to create RBD volume
+
+    """
+    def __init__(self, name=None, namespace='default'):
+        KubeVolume.__init__(self, name, namespace)
```
Comment (RazTamir) [inline ocs/volumes.py]: super().__init__(name, namespace) ?

File: ocs/volumes.py [hunk 22]
```diff
@@ -0,0 +1,257 @@
...
+        self.api_version = defaults.OPENSHIFT_REST_CLIENT_API_VERSION
+        self.service_sc = self.dyn_client.resources.get(
+            api_version=self.api_version,
+            kind=self.kind
+        )
+        self.template_path = os.path.join(TEMPLATE_DIR, "storageclass.yaml")
+
+    def create_storageclass(
+        self,
+        blockPool,
```
Comment (RazTamir) [inline ocs/volumes.py]: so this is a block storageClass?

File: ocs/volumes.py [hunk 23]
```diff
@@ -0,0 +1,257 @@
...
+        data = generate_yaml_from_jinja2_template_with_data(
+            self.template_path,
+            **cephblockpool_data
+        )
+        self.service_cbp.create(body=data, namespace=self.namespace)
+
+        return cbp_name
+
+
+class StorageClass(KubeVolume):
```
Comment (RazTamir) [inline ocs/volumes.py]: why storageClass inherit from Volume?

File: ocs/volumes.py [hunk 24]
```diff
@@ -0,0 +1,257 @@
...
+                accessmode="ReadWriteOnce ReadOnlyMany",
+                pvc_size=5
+            )
+
+        """
+        if self.name:
+            pvc_name = self.name
+        else:
+            pvc_name = f"{pvc_name_prefix}-{get_random_str()}"
+        pvc_size = f"{pvc_size}Gi"
```
Comment (RazTamir) [inline ocs/volumes.py]: That's really good!
I'm thinking if we can move this functionality directly to the jinja2 yaml.
something like - {{ size"Gi" | default(3Gi) }}
Not sure if that'll work

File: oc/openshift_ops.py (hunk 3 above)
Comment (vavuthu) [reply oc/openshift_ops.py]: After discussing with Shylesh, moved the implementaion cephblock pool to RookCluster.

File: oc/openshift_ops.py (hunk 5 above)
Comment (vavuthu) [reply oc/openshift_ops.py]: Done

File: oc/openshift_ops.py (hunk 6 above)
Comment (vavuthu) [reply oc/openshift_ops.py]: Done

File: oc/openshift_ops.py (hunk 7 above)
Comment (vavuthu) [reply oc/openshift_ops.py]: Done

File: ocs/volumes.py (hunk 21 above)
Comment (vavuthu) [reply ocs/volumes.py]: Done

File: ocs/volumes.py (hunk 22 above)
Comment (vavuthu) [reply ocs/volumes.py]: yes, that is block storage class.

File: ocs/volumes.py (hunk 24 above)
Comment (vavuthu) [reply ocs/volumes.py]: Used the same in yaml. Above variable gives the flexibility for user to define their own pvc size

File: ocs/rook.py [hunk 25]
```diff
@@ -138,3 +140,53 @@ def _get_pod_labels(self, pod_name, namespace):
...
+        cephblockpool_data = {}
+        cephblockpool_data['cephblockpool_name'] = cephblockpool_name
+        cephblockpool_data['rook_api_version'] = default.ROOK_API_VERSION
+        cephblockpool_data['failureDomain'] = failureDomain
+        cephblockpool_data['size'] = size
+
+        data = generate_yaml_from_jinja2_template_with_data(template_path, **cephblockpool_data)
+        service_cbp.create(body=data, namespace=namespace)
+
+        return cephblockpool_name
```
Comment (shylesh) [inline ocs/rook.py]: Do we need to return name here ? because it already came in as a parameter from caller .

File: ocs/volumes.py [hunk 26]
```diff
@@ -0,0 +1,252 @@
...
+    def __init__(self, name=None, namespace='default'):
+        super().__init__(name, namespace)
+        self.kind = kinds.CEPHBLOCKPOOL
+        self.api_version = defaults.OPENSHIFT_REST_CLIENT_API_VERSION
+        self.service_cbp = self.dyn_client.resources.get(
+            api_version=self.api_version,
+            kind=self.kind
+        )
+        self.template_path = os.path.join(TEMPLATE_DIR, "cephblockpool.yaml")
+        self.rk = RookCluster()
```
Comment (shylesh) [inline ocs/volumes.py]: TODO: RookCluster should be singleton. May be we can do it next PR.

File: ocs/volumes.py [hunk 27]
```diff
@@ -0,0 +1,252 @@
...
+                failureDomain="host",
+                size=3)
+
+        """
+        if self.name:
+            cbp_name = self.name
+        else:
+            cbp_name = f"{cephblockpool_name_prefix}-{get_random_str()}"
+
+        cbp_name = self.rk.create_cephblockpool(
```
Comment (shylesh) [inline ocs/volumes.py]: Do you want to assert pool creation ?

File: ocs/volumes.py (hunk 22 above)
Comment (shylesh) [reply ocs/volumes.py]: May be we can rename this class to BlockStorageClass or RBDStorageClass , not in this PR.

File: ocs/kinds.py [hunk 28]
```diff
@@ -0,0 +1,6 @@
+"""
+Hols all the 'kinds' required for OCS
```
Comment (RazTamir) [inline ocs/kinds.py]: Holds?

File: ocs/rook.py [hunk 29]
```diff
@@ -7,10 +7,12 @@
 """
```
Comment (RazTamir) [inline ocs/rook.py]: Please change the file name.
This has nothing to do with rook

File: ocs/rook.py [hunk 30]
```diff
@@ -138,3 +140,53 @@ def _get_pod_labels(self, pod_name, namespace):
...
+        namespace,
+        service_cbp,
+        failureDomain,
+        size
+    ):
+        """
+        Creates cephblock pool
+
+        Args:
+            cephblockpool_name (str): name of cephblockpool
```
Comment (RazTamir) [inline ocs/rook.py]: A capital letter at the beginning of the sentence - fix all

File: ocs/rook.py (hunk 25 above)
Comment (RazTamir) [reply ocs/rook.py]: +1 - returning the status is more reasonable

File: ocs/volumes.py [hunk 31]
```diff
@@ -0,0 +1,252 @@
...
+        size="3"
+    ):
+        """
+        Creates cephblock pool
+
+        Args:
+            cephblockpool_name_prefix (str): prefix given to cephblockpool
+            failureDomain (str): The failure domain across which the
+                                   replicas or chunks of data will be spread
+            size (int): The number of copies of the data in the pool.
```
Comment (RazTamir) [inline ocs/volumes.py]: The default you passed is str - "3"
And because this is not the volume size, maybe change it to replica_count

File: ocs/volumes.py (hunk 22 above)
Comment (RazTamir) [reply ocs/volumes.py]: +1 on renaming.
If we will have also cephfs storage class it will confuse ppl

Comment (vavuthu) [reply ocs/volumes.py]: Intention of keeping the create_storageclass is in future I would like to use the same for creating cephfs also by passing different parameter( I didn't try creating cepfs manually. At this point of time not sure we can use the same for both or not ). But if we are not going to use the same function for both , then I will change the names accordingly like create_storageclass_rbd and create_storageclass_cephfs . I will address this in another PR

File: ocs/kinds.py (hunk 28 above)
Comment (vavuthu) [reply ocs/kinds.py]: Done

File: ocs/rook.py (hunk 30 above)
Comment (vavuthu) [reply ocs/rook.py]: Done

File: ocs/volumes.py (hunk 31 above)
Comment (vavuthu) [reply ocs/volumes.py]: make sense to change to replica_count

Comment (vavuthu) [reply ocs/volumes.py]: Done

File: ocs/volumes.py (hunk 27 above)
Comment (vavuthu) [reply ocs/volumes.py]: It will assert it out by lower level function

File: ocs/volumes.py [hunk 32]
```diff
@@ -53,26 +49,29 @@ def __init__(self, name=None, namespace='default'):
...
+        replica_count="3"
     ):
         """
         Creates cephblock pool
 
         Args:
-            cephblockpool_name_prefix (str): prefix given to cephblockpool
+            cephblockpool_name_prefix (str): Prefix given to cephblockpool
             failureDomain (str): The failure domain across which the
                                    replicas or chunks of data will be spread
```
Comment (RazTamir) [inline ocs/volumes.py]: fix indentation:
```python
failureDomain (str): The failure domain across which the
    replicas or chunks of data will be spread
```

File: ocs/volumes.py [hunk 33]
```diff
@@ -81,7 +80,7 @@ def create_cephblockpool(
             create_cephblockpool(
                 cephblockpool_name_prefix="autotests-blockpool",
                 failureDomain="host",
-                size=3)
+                replica_count=3)
```
Comment (RazTamir) [inline ocs/volumes.py]: ) in new line

File: ocs/volumes.py (hunk 26 above)
Comment (vavuthu) [reply ocs/volumes.py]: will address in next PR

File: templates/ocs-deployment/cephblockpool.yaml (hunk 8 above)
Comment (vavuthu) [reply templates/ocs-deployment/cephblockpool.yaml]: changed to replica_count

File: ocs/volumes.py (hunk 32 above)
Comment (vavuthu) [reply ocs/volumes.py]: Done

File: ocs/volumes.py (hunk 33 above)
Comment (vavuthu) [reply ocs/volumes.py]: Done

File: ocs/rook.py (hunk 25 above)
Comment (vasukulkarni) [reply ocs/rook.py]: @vavuthu please resolve this one final review comment.

File: ocs/rook.py (hunk 29 above)
Comment (vasukulkarni) [reply ocs/rook.py]: This one is up for change in next PR

File: ocs/rook.py (hunk 25 above)
Comment (vavuthu) [reply ocs/rook.py]: Done. returning the status

**Comment by shylesh** [comment]
LGTM
